src/user/user.service.ts

```diff
@@ -202,4 +218,60 @@ export class UserService {
       throw new GlobalException(ErrorsTypes.USER_PROFILE_FAILED_TO_DELETE, error.message);
     }
   }
+
+  async addProfileImage(id: string, userId: string, image: Express.Multer.File): Promise<any> {
+    let existingAccommodation;
+    try {
+      existingAccommodation = await this.prismaService.userProfile.findUnique({
+        where: { id, userId },
+      });
+    } catch (error) {
+      throw new GlobalException(ErrorsTypes.USER_PROFILE_FAILED_TO_UPDATE, error.message);
+    }
+    if (!existingAccommodation) throw new NotFoundException(ErrorsTypes.NOT_FOUND_USER_PROFILE);
+
+    try {
+      const uploadedImagesResponse = await this.uploadImageToS3(image);
+
+      const imageToCreate = {
+        imageUrl: uploadedImagesResponse.data.imageUrl,
+        id,
+      };
+
+      const newUser = await this.prismaService.userProfile.update({
+        where: { id, userId },
+        data: imageToCreate,
+      });
+
+      return newUser;
+    } catch (error) {
+      throw new GlobalException(ErrorsTypes.USER_PROFILE_FAILED_TO_UPDATE, error.message);
+    }
+  }
+
+  async uploadImageToS3(file: Express.Multer.File): Promise<UploadImageResponse> {
+    const requestBody: UploadImageType = {
+      filename: this.generateRandomImageFileName(file.mimetype),
+      base64Image: file.buffer.toString('base64url'),
+    };
+
+    const { data } = await firstValueFrom(
+      this.httpService
+        .post(process.env.FILE_UPLOAD_URL!, requestBody, {
+          timeout: 20000,
+        })
+        .pipe(
+          catchError((error: AxiosError) => {
+            throw new GlobalException(ErrorsTypes.ACCOMMODATION_FAILED_TO_CREATE, error.message);
+          })
+        )
+    );
+    return data;
+  }
+
+  generateRandomImageFileName(mimetype: string) {
```

Comment:
use existing solution here, like nanoID, for example https://proglib.io/p/mozhet-li-nanoid-zamenit-uuid-2021-07-16